tests/dags/providers/provider_api_scripts/test_phylopic.py

```diff
@@ -1,23 +1,44 @@
 import json
 import logging
-import os
 from functools import partial
+from pathlib import Path
 from unittest.mock import patch
 
 import pytest
 from common.licenses import get_license_info
 from providers.provider_api_scripts import phylopic as pp
 
 
-RESOURCES = os.path.join(
-    os.path.abspath(os.path.dirname(__file__)), "resources/phylopic"
-)
+RESOURCES = Path(__file__).parent.resolve() / "resources/phylopic"
 
 logging.basicConfig(
     format="%(asctime)s - %(name)s - %(levelname)s:  %(message)s",
     level=logging.DEBUG,
 )
 
+IMAGE_FILESIZE = 100
+
+
+@pytest.fixture
+def image_data():
+    IMAGE_DATA_EXAMPLE = RESOURCES / "correct_meta_data_example.json"
+    with open(IMAGE_DATA_EXAMPLE) as f:
+        yield json.load(f)
+
+
+@pytest.fixture(autouse=True)
+def file_size_patch():
+    with patch(
+        "providers.provider_api_scripts.phylopic._get_filesize"
+    ) as _get_filesize_mock:
+        _get_filesize_mock.return_value = IMAGE_FILESIZE
+        yield
```

Comment:
Doesn't this kind just mask over the testing for the filesize logic a bit? Is there a way for us to mock the actual HEAD request response being made so that we can exercise the logic in `_get_filesize` instead of just mocking it (thereby never actually covering that function's implementation)?